recipes/autoconf/all/conanfile.py

```diff
@@ -13,110 +18,86 @@ class AutoconfConan(ConanFile):
     topics = ("autoconf", "configure", "build")
     license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
     settings = "os", "arch", "compiler", "build_type"
+    generators = "AutotoolsDeps", "AutotoolsToolchain"
+    win_bash = True
 
     exports_sources = "patches/*"
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
-
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
-    def requirements(self):
-        self.requires("m4/1.4.19")
+    def package_id(self):
+        self.info.clear()
 
     def build_requirements(self):
         if hasattr(self, "settings_build"):
             self.build_requires("m4/1.4.19")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self._settings_build.os == "Windows" and not environ.get("CONAN_BASH_PATH"):
+            self.tool_requires("msys2/cci.latest")
 
-    def package_id(self):
-        self.info.header_only()
+    def requirements(self):
+        self.requires("m4/1.4.19")
+
+    def layout(self):
+        basic_layout(self, src_folder="source")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        apply_conandata_patches(self)
+        autotool = Autotools(self)
+        autotool.configure()
+        autotool.make()
+        autotool.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])
```

Comment:
```suggestion
        autotool.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])
```

do not install files in `build()` method